README.md

```diff
@@ -44,35 +45,189 @@ Commonjs/`require` is also supported.
+`referencesTTL` value should be set at the maximum of all the `ttl`s, to let them be available for every cache entry, but at the same time, they expire, avoiding data leaking.  
+Anyway, we should keep `references` up-to-date to be more efficient on writes and invalidation, using the `garbage collector` function, that prunes the expired references: while expired references do not compromise the cache integrity, they slow down the I/O operations.  
+Storage `memory` doesn't have `gc`.
+
+### Redis garbage collector
+
+As said, While the garbage collector is optional, is highly recommended to keep references up to date and improve performances on setting cache entries and invalidation of them.  
+
+### `storage.gc([mode], [options])`
+
+* `mode`: `lazy` (default) or `strict`.
+  In `lazy` mode, only a chunk of the `references` are randomly checked, and probably freed; running `lazy` jobs tend to eventually clear all the expired `references`.
+  In `strict` mode, all the `references` are checked and freed, and after that, `references` and entries are perfectly clean.
+  `lazy` mode is the light heuristic way to ensure cached entries and `references` are cleared without stressing too much `redis`, `strict` mode at the opposite stress more `redis` to get a perfect result.
+  The best strategy is to combine them both, running often `lazy` jobs along with some `strict` ones, depending on the size of the cache.
+
+Options:
+
+* `chunk`: the chunk size of references analyzed per loops, default `64`
+* `lazy~chunk`: the chunk size of references analyzed per loops in `lazy` mode, default `64`; if both `chunk` and `lazy.chunk` is set, the maximum one is taken
+* `lazy~cursor`: the cursor offset, default zero; cursor should be set at `report.cursor` to continue scanning from the previous operation
+
+Return `report` of the `gc` job, as follows
+
+```json
+"report":{
+  "references":{
+      "scanned":["r:user:8", "r:group:11", "r:group:16"],
+      "removed":["r:user:8", "r:group:16"]
+  },
+  "keys":{
+      "scanned":["users~1"],
+      "removed":["users~1"]
+  },
+  "loops":4,
+  "cursor":0,
+  "error":null
+}
+```
+
+Example
+
+```js
+import { createCache, createStorage } from 'async-cache-dedupe'
+
+const cache = createCache({
+  ttl: 5,
+  storage: { type: 'redis', options: { client: redisClient, invalidation: true } },
+})
+// ... cache.define('fetchSomething'
+
+const storage = createStorage('redis', { client: redisClient, invalidation: true })
+
+let cursor
+setInterval(() => {
+  const report = await storage.gc('lazy', { lazy: { cursor }})
+  if(report.error) {
+    console.error('error on redis gc', error)
+    return
+  }
+  console.log('gc report (lazy)', report)
+  cursor = report.cursor
+}, 60e3).unref()
+
+setInterval(() => {
+  const report = await storage.gc('strict', chunk: 128})
+  if(report.error) {
+    console.error('error on redis gc', error)
+    return
+  }
+  console.log('gc report (strict)', report)
+}, 10 * 60e3).unref()
+
+```
+
```

Comment:
Please add a sentence saying that there is no need to run this job, however it will reduce the memory usage if it is used.